### ISSUE TITLE
Simplify and clean up worktrees form

### ISSUE THREAD

**Issue by KvanTTT** [gitextensions/gitextensions#8696]
The current form is not intuitive and overwhelmed in my opinion. I suggest implementing the following changes:

* Replace full paths with short paths (full paths are shown in tooltip on mouseover)
* Replace `Deleted` status with icons
* Strip `ref/heads/` prefix for branches
* Move `Open` and `Delete` buttons from DataGrid to form
* Add `Create` button

Also, I suggest replacing `Worktrees` menu item with `Create a worktree...` and `Manage worktrees...` submenus to get rid of excess navigation click.

## Current behaviour

![Current manage worktrees form](https://user-images.githubusercontent.com/1150330/103030119-3309d900-456c-11eb-87c3-b807903bd824.png)

## Expected behaviour

![Expected manage worktrees form](https://user-images.githubusercontent.com/1150330/103033484-99462a00-4573-11eb-9bd3-75a9baad6592.png)

## Environment

- Git Extensions 33.33.33
- Build 7df32bd123e58e52768f322cd29eb43d48714e1b (Dirty)
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.17763.0
- .NET Framework 4.8.4250.0
- DPI 144dpi (150% scaling)

**Comment by gerhardol** [comment]
Agree. It is not often I use the form, but it is sad to see it and the manage/create split is strange.
SHA can be hidden to, copy in context menu instead?

**Comment by KvanTTT** [comment]
> It is not often I use the form, but it is sad to see it and the manage/create split is strange.

I think some people don't use worktrees feature because the current UI is not very convenient and clear. I think the improvements in the issue and displaying worktrees at the left pane (#7048) can involve more users to use worktrees.

> SHA can be hidden to, copy in context menu instead?

Agree. It also can be shown via a tooltip.

**Comment by gerhardol** [comment]
> can involve more users to use worktrees.

discover worktrees is the first step

**Comment by RussKie** [comment]
> ![Expected manage worktrees form](https://user-images.githubusercontent.com/1150330/103033484-99462a00-4573-11eb-9bd3-75a9baad6592.png)

Few general comments on this screenshot:

1. I don't think an icon on its own is sufficiently clear indicator of the state. That is, I'm not finding the deleted state being obvious. It also won't be very accessible for users using accessibility tools (which is a big work on its own, but we shouldn't make it more difficult from onset).
Perhaps we could make it a big more obvious, and _this is a discussion point_ - e.g. make the text for deleted entries in different colour (e.g. grey?) and maybe crossed out too (?).
RE: colour - whilst in git world and in MS Office red is typically used to indicate a deletion, I am not sure it would work here, but certainly worth exploring it.
RE: crossing out - I think this is something universally used, so perhaps this metaphor will work here. Worth exploring. 
2. The datagridview visual presentation needs to be updated, e.g. background needs to be made white, etc. Please have a look at Scripts settings page
3. Do we need "Open" button? Can we replace it with a double click or a contextmenu instead? What does "Open" mean for a deleted worktree? 
4. "Add" / "Remove" - use the same icons as in FormRemotes.
5. "Add" and "Prune" button are global actions, where as "Open" and "Remove" are context actions, i.e. (I presume) they require a selection to be made. With this I'd think that "Add" and "Prune" buttons should be placed in the footer of the dialog, and the other buttons (if those are necessary) should be placed next to the datagridview. 
Please refer to https://github.com/gitextensions/gitextensions/issues/6183 (specifically https://github.com/gitextensions/gitextensions/pull/8510)

**Comment by KvanTTT** [comment]
Maybe move buttons from the bottom part to left (similar to FormRemotes and submodules)?

Also, I think it makes sense to add buttons `Lock` (`Unlock`), `Move`, maybe `Repair` (see detailed description on https://git-scm.com/docs/git-worktree#_commands).

How locked worktrees should be displayed in your opinion? Again, similar to FormRemotes with active and inactive repositories?

**Comment by RussKie** [comment]
I've never used worktrees, so I am not qualified to give opinions on what's useful (or not).
But IMO anything that is context aware should be next to the grid and exposed via a contextmenu, and anything that is a generally applicable operation without selecting a specific item - (likely) at the bottom of the dialog
